src/main/org/deidentifier/arx/metric/v2/MetricSDClassification.java

```diff
@@ -367,6 +457,9 @@ protected void initializeInternal(DataManager manager,
             int levels = hierarchy.getLevels();
             this.responseVariablesQIScaleFactors[i] = new double[levels];
             
+            // Obtain maximal generalization level
+            this.responseVariablesQIMaxGeneralization[i] = levels - 1;
```

Comment:
Is sort of redundant. There is not need to remove this, but you could use responseVariablesQIScaleFactors[i].length-1 instead...